mithril-common/src/cardano_block_scanner/chain_reader_block_streamer.rs

```diff
@@ -0,0 +1,320 @@
+            chain_reader_inner.set_chain_point(&from).await?;
+        }
+        Ok(Self {
+            chain_reader,
+            from,
+            until,
+            max_roll_forwards_per_poll: MAX_ROLL_FORWARDS_PER_POLL,
+            logger,
+        })
+    }
+
+    async fn get_next_chain_block_action(&self) -> StdResult<Option<BlockStreamerNextAction>> {
+        let mut chain_reader = self.chain_reader.try_lock()?;
+        match chain_reader.get_next_chain_block().await? {
+            Some(ChainBlockNextAction::RollForward { parsed_block }) => {
+                if parsed_block.block_number >= self.until {
+                    debug!(
+                        self.logger,
+                        "ChainReaderBlockStreamer received a RollForward above threshold block number ({})",
+                        parsed_block.block_number
+                    );
+                    Ok(None)
+                } else {
+                    debug!(
+                        self.logger,
+                        "ChainReaderBlockStreamer received a RollForward below threshold block number ({})",
+                        parsed_block.block_number
+                    );
+                    Ok(Some(BlockStreamerNextAction::ChainBlockNextAction(
+                        ChainBlockNextAction::RollForward { parsed_block },
+                    )))
+                }
+            }
+            Some(ChainBlockNextAction::RollBackward { rollback_point }) => {
+                debug!(
+                    self.logger,
+                    "ChainReaderBlockStreamer received a RollBackward({rollback_point:?})"
+                );
+                let block_streamer_next_action = if rollback_point == self.from {
+                    BlockStreamerNextAction::SkipToNextAction
+                } else {
+                    BlockStreamerNextAction::ChainBlockNextAction(
+                        ChainBlockNextAction::RollBackward { rollback_point },
+                    )
+                };
+                Ok(Some(block_streamer_next_action))
+            }
+            None => {
+                debug!(self.logger, "ChainReaderBlockStreamer received nothing");
+                Ok(None)
+            }
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::cardano_block_scanner::ScannedBlock;
+    use crate::chain_reader::FakeChainReader;
+    use crate::test_utils::TestLogger;
+
+    use super::*;
+
+    #[tokio::test]
+    async fn test_parse_expected_nothing_above_block_number_threshold() {
+        let logger = TestLogger::stdout();
+        let chain_reader = Arc::new(Mutex::new(FakeChainReader::new(vec![
+            ChainBlockNextAction::RollForward {
+                parsed_block: ScannedBlock::new("hash-1", 1, 10, 20, Vec::<&str>::new()),
+            },
+        ])));
+        let mut block_streamer =
+            ChainReaderBlockStreamer::try_new(chain_reader, None, 1, logger.clone())
```

Comment:
I'd remove `logger` to use `TestLogger::stdout()` directly as a parameter.
```suggestion
            ChainReaderBlockStreamer::try_new(chain_reader, None, 1, TestLogger::stdout()
```
Same suggestion for the other tests below.

Reply:
Done